v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -29,392 +29,585 @@ Note: More types will be added as needed.
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_006: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_007: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall copy the bytes of the `int64_t` value from the address pointed by `src_value` to `dst_value`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_007: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall copy the bytes of the `int64_t` value from the address pointed by `src_value` to `dst_value`. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall succeed and return 0. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall succeed and return 0. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).free
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).free
 
 ```c
 typedef int (*LOG_CONTEXT_PROPERTY_TYPE_FREE)(void* value);
 ```
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_009: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).free` shall return. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_009: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).free` shall return. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).get_type
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).get_type
 
 ```c
 typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
 ```
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).get_type` shall returns the property type `LOG_CONTEXT_PROPERTY_TYPE_int64_t`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).get_type` shall return the property type `LOG_CONTEXT_PROPERTY_TYPE_int64_t`. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string
+## LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)
+
+```c
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)(void* dst_value, int64_t src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)` copies the `src_value` value to the address pointed by `dst_value`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_081: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_082: [** `LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)` shall copy the `src_value` bytes of the `int64_t` to `dst_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_083: [** `LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)` shall succeed and return 0. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(int64_t)
+
+```c
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(int64_t)(int64_t src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(int64_t)` returns the number of bytes needed to hold `src_value`.
```

Comment:
> returns the number of bytes [](http://example.com/codeflow?start=55&length=28)

well - if it "returns the number of bytes" then it should return `size_t`.

Reply:
for strings this can fail and then we'd return a negative value.